.circleci/config.yml

```diff
@@ -0,0 +1,292 @@
+      build_and_small_tests:
+        parallelism: 1
+        machine:
+          image: ubuntu-1604:201903-01
+        parameters:
+          otp:
+            type: string
+            description: Version of the Erlang package to install
+          build_releases:
+            type: string
+            description: Flag to build release or not
+            default: "1"
+
+        environment:
+          SKIP_RELEASE: 1
+          SKIP_COV: 0
+          PRESET: small_tests
+        working_directory: ~/app
+
+        steps:
+          - checkout
+          - install_erlang:
+              otp: <<parameters.otp>>
+          - run:
+              name: Prepare for cache
+              command: |
+                ./tools/circleci-otp-to-str.sh <<parameters.otp>> > otp_version
+          - restore_cache:
+              key: build-cache-{{ .Branch }}-{{ .Revision }}--{{ checksum "otp_version" }}
+          - restore_cache:
+              key: deps-cache--{{ checksum "rebar.lock" }}--{{ checksum "big_tests/rebar.lock" }}--{{ checksum "otp_version" }}
+          - fetch_packages
+          - run:
+              name: Get deps
+              command: |
+                tools/configure with-all
+                ./rebar3 get-deps
+          - save_cache:
+              key: deps-cache--{{ checksum "rebar.lock" }}--{{ checksum "big_tests/rebar.lock" }}--{{ checksum "otp_version" }}
+              paths:
+                - ~/.cache/rebar3
+                - _build/default/lib
+          - run: ./rebar3 compile
+          - run:
+              name: Make Certs
+              command: |
+                make certs
+          - run:
+              name: Run Small tests
+              command: |
+                KEEP_COVER_RUNNING=1 ./tools/travis-test.sh -p $PRESET -s true
+          - run:
+              name: Install Releases
+              command: |
+                test 0 = <<parameters.build_releases>> || make devrel
+          - run:
+              name: Build Big Tests
+              command: |
+                tools/travis-build-tests.sh
+          - save_cache:
+              key: build-cache-{{ .Branch }}-{{ .Revision }}--{{ checksum "otp_version" }}
+              paths: ./
+
+
+      dialyzer:
+        parallelism: 1
+        machine:
+          image: ubuntu-1604:201903-01
+        working_directory: ~/app
+        parameters:
+          otp:
+            type: string
+            description: Version of the Erlang package to install
+        steps:
+          - checkout
+          - fetch_packages
+          - install_erlang:
+              otp: <<parameters.otp>>
+          - run:
+              name: Run Dialyzer
+              command: |
+                KEEP_COVER_RUNNING=1 ./tools/travis-test.sh -p dialyzer_only -s false
+                
+      package:
+        parallelism: 1
+        machine:
+          image: ubuntu-1604:201903-01
+        working_directory: ~/app
+        parameters:
+          platform:
+            type: string
```

Comment:
How about `Enum`? It would so much simpler to figure out what are possible values here.